gadget/gadget.go

```diff
@@ -533,7 +533,7 @@ func classicOrUndetermined(m Model) bool {
 }
 
 func wantsSystemSeed(m Model) bool {
-	return m != nil && m.Grade() != asserts.ModelGradeUnset
+	return m != nil && m.Grade() != asserts.ModelGradeUnset && !m.Classic()
```

Comment:
the changes to the gadget package will need their own tests and feel orthogonal to rest, they should be split out